src/test/java/org/mastodon/mamut/fitting/demo/ComputeMeanAndVarianceDemo.java

```diff
@@ -0,0 +1,117 @@
+package org.mastodon.mamut.fitting.demo;
+
+import net.imglib2.Cursor;
+import net.imglib2.img.Img;
+import net.imglib2.img.array.ArrayImgs;
+import net.imglib2.loops.LoopBuilder;
+import net.imglib2.type.numeric.real.FloatType;
+import net.imglib2.util.LinAlgHelpers;
+
+import org.mastodon.mamut.fitting.util.DemoUtils;
+import org.mastodon.mamut.fitting.util.MultiVariantNormalDistributionRenderer;
+import org.mastodon.mamut.model.Model;
+
+import java.util.Arrays;
+
+/**
+ * Computing the mean position and covariance matrix for a given segmented
+ * region of an image is an easy way to get good ellipsoid parameters for
+ * that segment.
+ * <p>
+ * Here is an example of how to do that.
+ */
+public class ComputeMeanAndVarianceDemo
+{
+	private static final int PIXEL_VALUE = 42;
+
+	public static void main( String[] args )
+	{
+		double[] center = { 40, 50, 60 };
+		double[][] givenCovariance = {
+				{ 400, 20, -10 },
+				{ 20, 200, 30 },
+				{ -10, 30, 100 }
+		};
+
+		Img< FloatType > image = generateExampleImage( center, givenCovariance );
+		double[] mean = computeMean( image );
+		double[][] computedCovariance = computeCovariance( image, mean );
+
+		System.out.println( "Given center: " + Arrays.toString( center ) );
+		System.out.println( "Computed mean: " + Arrays.toString( mean ) );
+		System.out.println( "Given covariance: " + Arrays.deepToString( givenCovariance ) );
+		System.out.println( "Computed covariance: " + Arrays.deepToString( computedCovariance ) );
+
+		Model model = new Model();
+		model.getGraph().addVertex().init( 0, mean, computedCovariance );
+		DemoUtils.showBdvWindow( DemoUtils.wrapAsAppModel( image, model ) );
+	}
+
+	/**
+	 * Returns an example image with a single ellipsoid. Pixel values are 0 or 42.
+	 * 0 is background, 42 is the ellipsoid.
+	 */
+	private static Img< FloatType > generateExampleImage( double[] center, double[][] cov )
+	{
+		Img< FloatType > image = ArrayImgs.floats( 100, 100, 100 );
+		MultiVariantNormalDistributionRenderer.renderMultivariateNormalDistribution( center, cov, image );
+		LoopBuilder.setImages( image ).forEachPixel( pixel -> {
+			if ( pixel.get() > 500 )
+				pixel.set( PIXEL_VALUE );
+			else
+				pixel.set( 0 );
+		} );
+		return image;
+	}
+
+	/**
+	 * Computes the mean position of the pixels whose value equals 42.
+	 */
+	private static double[] computeMean( Img< FloatType > image )
+	{
+		Cursor< FloatType > cursor = image.cursor();
+		double[] sum = new double[ 3 ];
+		double[] position = new double[ 3 ];
+		long counter = 0;
+		while ( cursor.hasNext() )
+			if ( cursor.next().get() == PIXEL_VALUE )
+			{
+				cursor.localize( position );
+				LinAlgHelpers.add( sum, position, sum );
+				counter++;
+			}
+		LinAlgHelpers.scale( sum, 1. / counter, sum );
+		return sum;
+	}
+
+	/**
+	 * Computes the covariance matrix of the pixels whose value equals 42.
+	 */
+	private static double[][] computeCovariance( Img< FloatType > image, double[] mean )
```

Comment:
And its the same for this method `int pixelValue` should be a parameter.

Reply:
Also here, the parameter would only be called with a constant and the method is private. Thus at the current state, it feels wrong to me to add a parameter that not really is a parameter.